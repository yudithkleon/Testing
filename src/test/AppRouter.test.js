import React from "react";
import { mount } from "enzyme";
import { AppRouter } from "../routers/AppRouter";
import { AuthContext } from "../auth/AuthContext";
// import '@testing-library/jest-dom'

describe("Pruebas de AppRouter", () => {
  const contextValue = {
    dispatch: jest.fn(),
    user: {
      logged: false,
    },
  };

  test("Mostrar login si no esta autenticado", () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  test("Mostrar el componente marvel Autenticado", () => {
    const contextValue = {
      dispatch: jest.fn(),
      user: {
        logged:true,
        name:'Yudith',
      },
    };

    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot({})
    expect( wrapper.find('.navbar').exists()).toBe(true);
  });
});
