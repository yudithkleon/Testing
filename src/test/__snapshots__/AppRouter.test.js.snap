// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de AppRouter Mostrar el componente marvel Autenticado 1`] = `Object {}`;

exports[`Pruebas de AppRouter Mostrar login si no esta autenticado 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      Login
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
